[SPARK-13364] Sort appId as num rather than str in history page.

zhuoliu opened the pull request:
## What changes were proposed in this pull request?

History page now sorts the appID as a string, which can lead to unexpected order for the case "application_11111_9" and "application_11111_20".
Add a new sort type called appId-numeric can fix it.
## How was the this patch tested?

This patch was manually tested with UI. See the screenshot below:
![sortappidbetter](https://cloud.githubusercontent.com/assets/11683054/13185564/7f941a16-d707-11e5-8fb7-0316368d3030.png)

Commit:
[SPARK-13364] Sort App Id column as num rather than str in History Server.

AmplabJenkins commented:
Can one of the admins verify this patch?

rxin commented:
Question: do you see the pull request template?

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/historypage.js:
```diff
@@ -37,6 +37,22 @@ function formatDuration(milliseconds) {
 }
 
+function makeIdNumeric(id) {
+  var strs = id.split("-");
```
The application ids are actually "_".  However if running in local mode then it is "-". local-1455658099053.  We should check is any other formats.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/historypage.js:
```diff
@@ -37,6 +37,22 @@ function formatDuration(milliseconds) {
+  var appSeqNum = strs[2];
+  var resl = strs[0] + "-" + strs[1] + "-";
+  var diff = 10 - appSeqNum.length;
+  while (diff > 0) {
```
we should at least comment this.  If java script has a function to pad or print strings it certain length it would be better.

Commit:
Address comments.

zhuoliu commented:
Hi @rxin , yes, I see the pull request template and now updated the description.
@tgravescs , I addressed your comment. Not find a javascript function I can directly call to padding zeros yet.

tgravescs commented:
Jenkins, test this please

SparkQA commented:
**[Test build #51673 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/51673/consoleFull)** for PR 11259 at commit [`94a7f78`](https://github.com/apache/spark/commit/94a7f78cfc862fc05599af4710fd2b62821fd945).

SparkQA commented:
**[Test build #51673 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/51673/consoleFull)** for PR 11259 at commit [`94a7f78`](https://github.com/apache/spark/commit/94a7f78cfc862fc05599af4710fd2b62821fd945).
 * This patch **fails from timeout after a configured wait of \`250m\`**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/51673/
Test FAILed.

zhuoliu commented:
Jenkins, test this please

tgravescs commented:
Jenkins, test this please

SparkQA commented:
**[Test build #51769 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/51769/consoleFull)** for PR 11259 at commit [`94a7f78`](https://github.com/apache/spark/commit/94a7f78cfc862fc05599af4710fd2b62821fd945).

SparkQA commented:
**[Test build #51769 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/51769/consoleFull)** for PR 11259 at commit [`94a7f78`](https://github.com/apache/spark/commit/94a7f78cfc862fc05599af4710fd2b62821fd945).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/51769/
Test PASSed.

tgravescs commented:
+1
